common/frame.go

```diff
@@ -147,6 +148,18 @@ func (f *Frame) inflightRequestsLen() int {
 	return len(f.inflightRequests)
 }
 
+func (f *Frame) getInflightRequest() map[network.RequestID]bool {
+	f.inflightRequestsMu.Lock()
+	defer f.inflightRequestsMu.Unlock()
```

Comment:
What about using `RLock` and `RUnlock` since we're only reading from `f.inflightRequests` and not changing it?

Reply:
Resolved in: 4d13d8a11e865be2ac6bd2ddd0b556fc4b0489be